config/types/config.go

```diff
@@ -22,15 +22,16 @@ import (
 )
 
 type Config struct {
-	Ignition Ignition `yaml:"ignition"`
-	Storage  Storage  `yaml:"storage"`
-	Systemd  Systemd  `yaml:"systemd"`
-	Networkd Networkd `yaml:"networkd"`
-	Passwd   Passwd   `yaml:"passwd"`
-	Etcd     *Etcd    `yaml:"etcd"`
-	Flannel  *Flannel `yaml:"flannel"`
-	Update   *Update  `yaml:"update"`
-	Docker   *Docker  `yaml:"docker"`
+	Ignition  Ignition   `yaml:"ignition"`
+	Storage   Storage    `yaml:"storage"`
+	Systemd   Systemd    `yaml:"systemd"`
+	Networkd  Networkd   `yaml:"networkd"`
+	Passwd    Passwd     `yaml:"passwd"`
+	Etcd      *Etcd      `yaml:"etcd"`
+	Flannel   *Flannel   `yaml:"flannel"`
+	Update    *Update    `yaml:"update"`
+	Docker    *Docker    `yaml:"docker"`
+	Locksmith *Locksmith `yaml:"locksmith"`
```

Comment:
I originally made the tweak in Andrew's PR that added etcd and flannel, the validation logic was choking when the section was omitted because `""` isn't a valid version for etcd. Making the value `nil`-able by making it a pointer made the validation logic for the section not run when the section was omitted.